apps/extension/src/routes/offscreen/build.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  InternalMessageHandler,
+  InternalResponse,
+} from '@penumbra-zone/types/src/internal-msg/shared';
+import { ActionBuildMessage } from './types';
+import {
+  Action,
+  ActionPlan,
+  TransactionPlan,
+  WitnessData,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1alpha1/transaction_pb';
+
+const spawnWorker = (
+  transactionPlan: TransactionPlan,
+  actionPlan: ActionPlan,
+  witness: WitnessData,
+  fullViewingKey: string,
+  key_type: string,
+): Promise<Action> => {
+  return new Promise((resolve, reject) => {
+    const worker = new Worker(new URL('../web-worker/router.ts', import.meta.url));
+
+    // Set up event listener to recieve messages from the web worker
```

Comment:
I considered using comlink, but it wasn't working properly! Let's stick to message-passing for simplicity. 

In terms of memory leaks, I've considered this and made some local changes that terminate the worker (`worker.terminate()`) and event listener (`worker.removeEventListener('message', handleMessage)`) to clean up the resources associated with the workers lifecycle.

Reply:
we actually already terminate the worker with `close()` inside `web-worker.ts `, but you're right, I still need to handle the event listener to prevent potential memory leaks.